src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs

```diff
@@ -0,0 +1,129 @@
+	/// </summary>
+	[Cmdlet(VerbsCommon.Set, "AzureRmSqlServerDnsAlias",
+		SupportsShouldProcess = true,
+ConfirmImpact = ConfirmImpact.Medium)]
+	[OutputType(typeof(Model.AzureSqlServerDnsAliasModel))]
+	public class SetAzureSqlServerDnsAlias : AzureSqlCmdletBase<IEnumerable<AzureSqlServerDnsAliasModel>, AzureSqlServerDnsAliasAdapter>
+	{
+		/// <summary>
+		/// Template to generate the Server Dns Alias Id
+		/// </summary>
+		public static string ServerDnsAliasIdTemplate = "/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Sql/servers/{2}/dnsAliases/{3}";
+
+		/// <summary>
+		/// Gets or sets the name of the Azure Sql Server Dns Alias
+		/// </summary>
+		[Parameter(Mandatory = true,
+			HelpMessage = "The Azure Sql Server Dns Alias name.")]
+		[Alias("Name")]
+		[ValidateNotNullOrEmpty]
+		public string DnsAliasName { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of the new server to which alias should point
+		/// </summary>
+		[Parameter(Mandatory = true,
+			HelpMessage = "The name of Azure Sql Server to which alias should point.")]
+		[ValidateNotNullOrEmpty]
+		public string TargetServerName { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of the new server to which alias is currently pointing
+		/// </summary>
+		[Parameter(Mandatory = true,
+			HelpMessage = "The name of Azure Sql Server to which alias is currently pointing.")]
+		[ValidateNotNullOrEmpty]
+		public string SourceServerName { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of the resource group of the old server
+		/// </summary>
+		[Parameter(Mandatory = true,
+			HelpMessage = "The name of resource group of the source server.")]
+		[ValidateNotNullOrEmpty]
+		public string SourceServerResourceGroupName { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of the subscription id of the old server
+		/// </summary>
+		[Parameter(Mandatory = true,
+			HelpMessage = "The subscription id of the source server")]
+		[ValidateNotNullOrEmpty]
+		public Guid SourceServerSubscriptionId { get; set; }
+
+		/// <summary>
+		/// Get entity method - nothing to return
+		/// </summary>
+		protected override IEnumerable<Model.AzureSqlServerDnsAliasModel> GetEntity()
+		{
+			return new List<Model.AzureSqlServerDnsAliasModel>();
+		}
+
+		/// <summary>
+		/// Apply user input to model - nothing to do
+		/// </summary>
+		protected override IEnumerable<Model.AzureSqlServerDnsAliasModel> ApplyUserInputToModel(IEnumerable<Model.AzureSqlServerDnsAliasModel> model)
+		{
+			return model;
+		}
+
+		/// <summary>
+		/// Persist changes - actually invoke acquire on alias to move it from one server to another
+		/// </summary>
+		protected override IEnumerable<Model.AzureSqlServerDnsAliasModel> PersistChanges(IEnumerable<Model.AzureSqlServerDnsAliasModel> entity)
+		{
+			return new List<Model.AzureSqlServerDnsAliasModel>()
+			{
+				ModelAdapter.AcquireServerDnsAlias(
+					this.ResourceGroupName,
```

Comment:
oh I see, there is base .ResourceGroupName. Seems pretty confusing that this goes with TargetServerName.

Reply:
I have added alias TargetResourceGroupName for ResourceGroupName